vng_api_common/audittrails/viewsets.py

```diff
@@ -30,13 +30,13 @@ def create_audittrail(self, status_code, action, version_before_edit, version_af
             main_object = self.get_audittrail_main_object_url(data, self.audit.main_resource)
 
         try:
-            applicatie_id = self.request.jwt_payload['client_id']
-        except:
+            applicatie_id = self.request.jwt_auth.client_id
+        except AttributeError:
```

Comment:
This will never raise `AttributeError`, but `PermissionDenied` error. However, that should have hit earlier in the auth chain already, leading to a shortcut HTTP 403 response.

I think a better approach is:

```python
applications = self.request.jwt_auth.applicaties
if len(applications) > 1:
    logger.warning("Unexpectedly found %d applications, expected at most one", len(applications))
if applications:
    application = applications[0]
    app_id, app_presentation = str(application.uuid), application.label
else:
    header = 'X_NLX_REQUEST_APPLICATION_ID'
    if hasattr(self.request, 'headers'):
        app_id = self.request.headers.get(header)
    else:
        app_id = self.request.META.get(header)
    app_presentation = app_id  # we don't have any extra information...

```